### ISSUE TITLE
recommandation mapping 

### ISSUE THREAD

**Issue by ayoubelmhamdi** [L3MON4D3/LuaSnip#96]
you should be add this lines in `readme` file to default configuration
because i spend much time to configure luasnip ,  but it's easy
```lua
local function prequire(...)
    local status, lib = pcall(require, ...)
    if (status) then return lib end
    return nil
end

local luasnip = prequire('luasnip')

local t = function(str)
    return vim.api.nvim_replace_termcodes(str, true, true, true)
end

local check_back_space = function()
    local col = vim.fn.col('.') - 1
    if col == 0 or vim.fn.getline('.'):sub(col, col):match('%s') then
        return true
    else
        return false
    end
end

_G.tab_complete = function()
    if vim.fn.pumvisible() == 1 then
        return t "<C-n>"
    elseif luasnip and luasnip.expand_or_jumpable() then
        return t "<Plug>luasnip-expand-or-jump"
    elseif check_back_space() then
        return t "<Tab>"
    else
        return vim.fn['compe#complete']()
    end
end
_G.s_tab_complete = function()
    if vim.fn.pumvisible() == 1 then
        return t "<C-p>"
    elseif luasnip and luasnip.jumpable(-1) then
        return t "<Plug>luasnip-jump-prev"
    else
        return t "<S-Tab>"
    end
end


vim.api.nvim_set_keymap("i", "<Tab>", "v:lua.tab_complete()", {expr = true})
vim.api.nvim_set_keymap("s", "<Tab>", "v:lua.tab_complete()", {expr = true})
vim.api.nvim_set_keymap("i", "<S-Tab>", "v:lua.s_tab_complete()", {expr = true})
vim.api.nvim_set_keymap("s", "<S-Tab>", "v:lua.s_tab_complete()", {expr = true})

```

**Comment by ayoubelmhamdi** [comment]
and why not add default conf of `compe` to `redme` file , it;s be easy for beginner

**Comment by ayoubelmhamdi** [comment]
and why not add one snipe for test `luasnip` is it work property

**Comment by L3MON4D3** [comment]
That's the code from #1, right? I thought that linking the issue would be enough, but if it wasn't for you, maybe there are others who feel the same. I'll include it.
As for testing, you could `:luafile` `Examples/snippets.lua` which allows users to get (at least a bit) acquainted with snippets in lua, I'll mention that too.
Thank you👍

**Comment by ayoubelmhamdi** [comment]
yes i found config for compe in  #1  after lot of time and i found treu keymap in [How to use tab to navigate completion menu?](https://github.com/hrsh7th/nvim-compe#how-to-use-tab-to-navigate-completion-menu)

**Comment by ayoubelmhamdi** [comment]
add this line to redme or documanation
for source `vscode` snippet (like [friendly](https://github.com/rafamadriz/friendly-snippets") )automatique and not be duplicate 
```
augroup testgroup
    autocmd!
    autocmd FileType * :lua require('luasnip.loaders.from_vscode').load()
augroup END

```

**Comment by L3MON4D3** [comment]
> add this line to redme or documanation
for source vscode snippet (like friendly ) automatique and not be duplicate

That function is mentioned in `Examples/snippets.lua`, but I think we ought to fix it instead of directing users to use another function that has different capabilities.

**Comment by L3MON4D3** [comment]
Fixed with 69ac54c
